probe-rs/src/probe/jlink/mod.rs

```diff
@@ -550,7 +561,7 @@ impl JLink {
 
         self.require_interface_supported(intf)?;
 
-        self.write_cmd(&[Command::SelectIf as u8, intf as u8])?;
+        self.write_cmd(Command::SelectIf, &[Command::SelectIf as u8, intf as u8])?;
```

Comment:
it's a bit strange that you have to duplicate the command. perhaps write_cmd could take command+args and prepend the command byte itself?

Reply:
I've thought about it, but I wanted to keep this reasonably efficient (i.e. not cloning the whole swd/jtag transfer buffers), some uses send a `&[u8]` buffer so I can't prepend to them with const-generics, and I've also tried just setting the first byte of the buffer to 0 but that's easy to get wrong.

Decoding the first byte is a nightmare.

I wasn't able to find any solutions here that I actually liked and this one seems to have the fewest downsides.